### PR TITLE
feat: Add Support For time based partitioning in `overwrite` write_mode 

### PR DESCRIPTION
**Pull request by bbernays**
#### Summary

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: schema/column.go [hunk 1]
```diff
@@ -43,6 +43,8 @@ type Column struct {
 	// If IgnoreInTests is true, verification is skipped for this column.
 	// Used when it is hard to create a reproducible environment with this column being non-nil (e.g. various error columns).
 	IgnoreInTests bool `json:"-"`
+
+	Skippable bool `json:"-"`
```
Comment (disq) [inline schema/column.go:47]: Leaked from somewhere else

File: plugins/destination/managed_writer.go [hunk 2]
```diff
@@ -160,6 +160,10 @@ func (p *Plugin) writeManagedTableBatch(ctx context.Context, sourceSpec specs.So
 		if len(r.Data) < len(tables.Get(r.TableName).Columns) {
 			r.Data = append([]schema.CQType{sourceColumn, syncTimeColumn}, r.Data...)
 		}
+		colIndex := tables.Get(r.TableName).Columns.Index(schema.CqSyncTimeColumn.Name)
+		if p.spec.PartitionMinutes > 0 && colIndex != -1 {
+			_ = r.Data[colIndex].Set(r.Data[colIndex].Get().(time.Time).UTC().Round(time.Duration(p.spec.PartitionMinutes) * time.Minute))
```
Comment (disq) [inline plugins/destination/managed_writer.go:165]: Can this get any longer? 😂  maybe make it into a helper fn? Repeated in plugin.go too

Comment (bbernays) [reply plugins/destination/managed_writer.go]: Broke it into a standalone func

**Review by erezrokah — COMMENTED** [review]
Added a couple of comments and will need to test this too a bit

**Comment by yevgenypats** [comment]
We are moving transformations now to a separate step, right now it's part of the CLI.

**Pull request closed without merging**